src/main/scala/sbtcrossproject/CrossProject.scala

```diff
@@ -1,184 +1,115 @@
+  def overrideConfigs(cs: Configuration*): CrossProject[A] =
+    copy(projects.map((x) => x._1 -> x._2.overrideConfigs(cs: _*)))
 
-  def overrideConfigs(cs: Configuration*): CrossProject =
-    mapProjects(_.overrideConfigs(cs: _*))
+  private def copy[T <: Platform](
+      projects: Map[Platform, Project] = projects): CrossProject[T] =
+    new CrossProject(id, root, projects)
 
-  /** Configures how settings from other sources, such as .sbt files, are
-   *  appended to the explicitly specified settings for this project.
-   *
-   *  Note: If you disable AutoPlugins here, non-JVM platforms will not work
-   */
-  def settingSets(select: AddSettings*): CrossProject =
-    mapProjects(_.settingSets(select: _*))
+  def dependsOn(deps: CrossClasspathDependency*): CrossProject[A] =
+    copy(
+        projects.map((x) => x._1 -> x._2.dependsOn(deps.map(_.dep(x._1)): _*)))
 
-  def settings(ss: Def.Setting[_]*): CrossProject =
-    mapProjects(_.settings(ss: _*))
+  def crossType(ct: CrossType): CrossProject[A] = {
+    copy(ct.modify(root, projects))
+  }
 
-  override def toString(): String = {
+  override def toString: String = {
     val entries = this.projects.map {
-      case (platform, project) => "$platform = $project"
+      case (t, p) => s"${t.name} -> $p"
     }
-    s"CrossProject(${entries.mkString(", ")})"
+    s"CrossProject(${entries.mkString(",\n")})"
   }
-
-  // Helpers
-
-  private def mapCrossPlatforms(transform: CrossPlatform => Project): CrossProject =
-    copy(projects.map { case (p, _) => p -> transform(p) })
-
-  private def mapProjects(transform: Project => Project): CrossProject =
-    copy(projects.mapValues(transform))
-
-  private def copy(projects: Map[CrossPlatform, Project] = projects): CrossProject =
-    new CrossProject(crossType, projects)
 }
 
 object CrossProject extends CrossProjectExtra {
-  def apply(id: String, platforms: Seq[CrossPlatform], base: File,
-      crossType: CrossType): CrossProject =
-    apply(platforms.map(p => (p, id + p.name)).toMap, base, crossType)
-
-  def apply(ids: Map[CrossPlatform, String], base: File,
-      crossType: CrossType): CrossProject = {
-
-    val sss = sharedSrcSettings(crossType)
-
-    val projects = ids.map { case (p, id) =>
-      val project = Project(id, crossType.dir(p, base)).
-        settings(sss: _*).
-        enablePlugins(p.plugin.toSeq: _*)
-
-      p -> project
-    }
-
-    new CrossProject(crossType, projects)
-  }
 
-  private def sharedSrcSettings(crossType: CrossType) = Seq(
-      unmanagedSourceDirectories in Compile ++= {
-        makeCrossSources(crossType.sharedSrcDir(baseDirectory.value, "main"),
-            scalaBinaryVersion.value, crossPaths.value)
-      },
-      unmanagedSourceDirectories in Test ++= {
-        makeCrossSources(crossType.sharedSrcDir(baseDirectory.value, "test"),
-            scalaBinaryVersion.value, crossPaths.value)
-      }
-  )
-
-  // Inspired by sbt's Defaults.makeCrossSources
```

Comment:
@sjrd please review the last version,analog of this code was moved to toCrossProject method